aspnetcore/tutorials/first-mongo-app.md

```diff
@@ -0,0 +1,167 @@
+
+
+### Set up MongoDB
+
+To get started, install MongoDB on your system. You can download MongoDB from [here](https://docs.mongodb.com/manual/administration/install-community/). If you're using Windows, MongoDB is installed in the following path:
+
+C:\Program Files\MongoDB
+
+Add *C:\Program Files\MongoDB\Server\3.4\bin* to the `Path` environment variable to access MongoDB from anywhere.
+
+### MongoDB implementation
+
+To use MongoDB, we have to create a database, make collections, and store documents in it. To do that, create a a data folder where your data will be stored. You can choose any location on your system as the data folder. (Keep note that the MongoDB CLI doesn't create new directories).
+
+Open a command shell:
+
+```console
+mongod --dbpath <enter-the-path>
+```
+
+This will connect to MongoDB on port 27017.
+
+Open another command shell instance and run the following:	
+
+```console
+mongo
+```
+
+This connects to the default test database. Run the following command in command shell:
+
+```console
+useBookstoreDb
+```
+
+If it doesn't already exist, a database named *BookstoreDb* is created. If the database does exist, its connection is opened for transactions.
+
+We can create a collection using following command:
+
+```console
+db.createCollection('Books')
+```
+
+The schema for the `Books` collection can be defined with the following command:
+
+```console
+db.Books.insert({'BookId':1,'BookName':'Design Patterns','Price':3000,'Category':'Computers', 'Author':' Ralph Johnson'})
+```
+
+To see the entries in your database, you can use following command:
+
+```console
+db.Books.find({})
+```
+
+The following result will be displayed:
+
+```console
+db.Books.find({})
+{ "_id" : ObjectId("5ad08f91a4f1c236ef1b4fff"), "BookId" : 1, "BookName" : "Design Patterns", "Price" : 3000, "Category" : "Computers", "Author" : " Ralph Johnson" }
+```
+
+The schema will add `_id` property. This property will be an `ObjectId` which will be generated automatically.
+
+Now that our database is ready, we can start creating our web API using .NET Core.
+
+### Create the ASP.NET Core web API
+
+You need Visual Studio with .NET Core SDK for creating the .NET app. For the purpose of this article, I will be using Visual Studio 2017.
+
+In visual studio, go to **File** > **New** > **Project** and select ASP.NET Core Web app from .NET Core tab.
+
+![New Project](~/tutorials/first-mongo-app/_static/newProject.png)
+
+Name the project "BookMongo" (or whatever you like) and click OK. This will open a window for selecting templates. Choose the **API** template and click OK.
+
+![Select Template](~/tutorials/first-mongo-app/_static/apiTemplate.png)
+
+This will create your web API project.
+
+Right-click the project in Solution Explorer and select Manage NuGet Packages.
+
+Search for *mongocsharpdriver* and install the package.
```

Comment:
This is the legacy driver. Can you get this tutorial working with MongoDB.Driver instead?

Reply:
Sure Scott, I'll work on updating the tutorial ✌🏻

Reply:
Thank you for all your work on this so far!